File: Data Structures & Algorithms/max-water-container/submission-2.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l,r=0,len(heights)-1
        res=0
        while l<r:
            vol = (r-l)*min(heights[l],heights[r])
            res=max(vol,res)
            if(heights[l]<heights[r]):
                l+=1
            else:
                r-=1
        return res

File: Data Structures & Algorithms/task-scheduling/submission-2.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        vector<int> cnt(26,0);
        for(char task :tasks){
            cnt[task-'A']++;
        }
        sort(cnt.begin(),cnt.end());
        int maxf=cnt[25];
        int idle = (maxf-1)*n;
        for(int i=24;i>=0;i--){
            idle-=min(maxf-1,cnt[i]);
        }
        return max(0,idle)+tasks.size();
        
    }
};
